api/user.js

```diff
@@ -0,0 +1,52 @@
+import { BAD_REQUEST, TYPE_JSON } from './defs.js'
+import { POST } from './router.js'
+import * as db from './db.js'
+
+// function to check the form's data
+const verifyInformations = (body) => {
+  let checks = []
+  for (const [name, item] of Object.entries(body)) {
+    switch (name) {
+      case 'own_contact':
+        isNaN(item) && checks.push('please enter your phone number')
+        break
+      case 'emergency_contact':
+        isNaN(item) &&
+          checks.push(
+            'please enter a phone number to join in case of emergency',
+          )
+        break
+      default:
+        break
+    }
+  }
+  return checks
+}
```

Comment:
All right so, it's good that you return multiple errors if they are many, but I think we should keep the source of the error:
instead of returning `[ 'please enter a phone number to join in case of emergency' ]` we could return
`{ emergency_contact: 'please enter a phone number to join in case of emergency' }`

this way we preserve the info of what key failed, we can use that to give inline errors back to the client